python/sparkdl/graph/graph_builder.py

```diff
@@ -0,0 +1,252 @@
+        else:
+            gdef = self.graph.as_graph_def(add_shapes=True)
+        return GraphFunction(graph_def=gdef,
+                             input_names=[tfx.validated_input(self.graph, elem) for elem in inputs],
+                             output_names=[tfx.validated_output(self.graph, elem) for elem in outputs])
+
+    def import_graph_function(self, gfn, input_map=None, name="GFN-IMPORT", **gdef_kargs):
+        """
+        Import a GraphFunction object into the current session
+
+        :param gfn: GraphFunction, an object representing a TensorFlow graph and its inputs and outputs
+        :param input_map: dict, mapping from input names to existing graph elements
+        :param name: str, the scope for all the variables in the GraphFunction's elements
+        :param gdef_kargs: other keyword elements for TensorFlow's `import_graph_def`
+        """
+        try:
+            del gdef_kargs["return_elements"]
+        except KeyError:
+            pass
+        if input_map is not None:
+            assert set(input_map.keys()) <= set(gfn.input_names), \
+                "cannot locate provided input elements in the graph"
+
+        input_names = gfn.input_names
+        output_names = gfn.output_names
+        if name is not None:
+            name = name.strip()
+            if len(name) > 0:
+                output_names = [
+                    name + '/' + op_name for op_name in gfn.output_names]
+                input_names = [
+                    name + '/' + op_name for op_name in gfn.input_names]
+
+        # When importing, provide the original output op names
+        tf.import_graph_def(gfn.graph_def,
+                            input_map=input_map,
+                            return_elements=gfn.output_names,
+                            name=name,
+                            **gdef_kargs)
+        feeds = [tfx.get_tensor(self.graph, name) for name in input_names]
+        fetches = [tfx.get_tensor(self.graph, name) for name in output_names]
+        return (feeds, fetches)
+
+
+class GraphFunction(object):
+    """
+    Represent a TensorFlow graph with its GraphDef, input and output operation names.
+
+    :param graph_def: GraphDef, a static ProtocolBuffer object holding informations of a TensorFlow graph
+    :param input_names: names to the input graph elements (must be of Placeholder type)
+    :param output_names: names to the output graph elements
+    """
+
+    def __init__(self, graph_def, input_names, output_names):
+        """
+        :param graph_def: GraphDef object
+        :param input_names: list of input (operation) names (must be typed `Placeholder`)
+        :param output_names: list of output (operation) names
+        """
+        self.graph_def = graph_def
+        self.input_names = input_names
+        self.output_names = output_names
+
+    def dump(self, fpath):
+        """
+        Store the GraphFunction to a file
```

Comment:
I actually changed this to explicitly store the `GraphDef` as bytes and the filename, inputs and outputs names in a JSON file. 
https://github.com/databricks/spark-deep-learning/blob/1dda1ef8c9d7ff561ee9bc85cbcddf9e4095466b/python/sparkdl/graph/builder.py#L140